src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,63 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const _ = require("lodash");
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["msg", "say", "query"];
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = cmd === "say" ? chan.name : args.shift();
+
+	if (cmd === "query") {
+		if (args.length === 0 || target.length === 0) {
+			chan.pushMessage(this, new Msg({
+				type: Msg.Type.ERROR,
+				text: "You cannot open a query window without an argument.",
+			}));
+			return;
+		}
+
+		const query = _.find(network.channels, {name: target});
+
+		if (typeof query !== "undefined") {
+			return;
```

Comment:
This return means if there's a query open already, the message will not be sent.